easybuild/easyconfigs/p/PETSc/PETSc-3.17.0-foss-2021a.eb

```diff
@@ -0,0 +1,84 @@
+# Updated to version 3.17.0 with significantly more modules added
+# J. Sassmannshausen ICL/UK 
+
+easyblock = 'ConfigureMake'
+
+name = 'PETSc'
+version = '3.17.0'
+
+homepage = 'https://www.mcs.anl.gov/petsc'
+description = """PETSc, pronounced PET-see (the S is silent), is a suite of data structures and routines for the
+ scalable (parallel) solution of scientific applications modeled by partial differential equations.
+
+This installation is build with significant amounts of modules in it compared to previous versions."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+toolchainopts = {'openmp': True, 'usempi': True, 'pic': True}
+
+source_urls = [
+    'https://ftp.mcs.anl.gov/pub/petsc/release-snapshots/',
+    'ftp://ftp.mcs.anl.gov/pub/petsc/release-snapshots/',
+]
+sources = [SOURCELOWER_TAR_GZ]
+patches = [
+    'webclient-3.17.0.patch',
+]
+checksums = [
+    '96d5aca684e1ce1425891a620d278773c25611cb144165a93b17531238eaaf8a',  # petsc-3.17.0.tar.gz
+    '2ba253df42853385e2683dbc87dfcb66d4ce4bcd84e054efa8f466274f49bc81',  # webclient-3.17.0.patch
+]
+
+builddependencies = [('CMake', '3.20.1')]
+
+dependencies = [
+    ('Python', '3.9.5'),
+    ('SciPy-bundle', '2021.05'),
+    ('Boost', '1.76.0'),
+    ('METIS', '5.1.0'),
+    ('SCOTCH', '6.1.0'),
+    ('MUMPS', '5.4.0', '-metis'),
+    ('SuiteSparse', '5.10.1', '-METIS-5.1.0'),
+    ('Hypre', '2.21.0'),
+    ('zlib', '1.2.11'),
+    ('BLIS', '0.8.1'),
+    ('Szip', '2.1.1'),
+    ('HDF5', '1.12.1'),
+    ('Eigen', '3.3.9'),
```

Comment:
Eigen should only need to be a build dependency (since it's header-only). I imagine googletest isn't a runtime requirement either

Reply:
True. I forgot about `Eigen` being only header. I was quite happy that it all worked after 2 days of playing with it to be honest. :-)